lib/src/main/java/com/auth0/android/lock/Lock.java

```diff
@@ -150,8 +151,8 @@ private void processEvent(@NonNull Context context, @NonNull Intent data) {
         switch (action) {
             case Constants.AUTHENTICATION_ACTION:
                 Log.v(TAG, "AUTHENTICATION action received in our BroadcastReceiver");
-                if (data.hasExtra(Constants.ERROR_EXTRA)) {
-                    callback.onError(new LockException(data.getStringExtra(Constants.ERROR_EXTRA)));
+                if (data.hasExtra(Constants.EXCEPTION_EXTRA)) {
```

Comment:
The former was not used from the code. Replaced with a more meaningful extra that can hold the Exception instance with all the details.